File: SelectionSort/selection.cpp
#include <iostream>

#define cout std::cout
#define cin std::cin
#define endl std::endl

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

void selectionSort(int arr[], int n) {
    for(int i = 0; i < n; i++) {
        int min = i;
        for(int j = i+1; j < n; j++) {
            if(arr[j] < arr[min]) {
                min = j;
            }
        }
        swap(arr[min], arr[i]);
    }
}

int main() {
    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    int arr[n];

    for(int i = 0; i < n; i++) {
        cout << "Enter an element: ";
        cin >> arr[i];
    }

    selectionSort(arr, n);

    cout << "Sorted array: " << endl;

    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
}

File: SelectionSort/selection.lua
function selectionSort(arr)
    local n = #arr

    for i = 1, n do
        local min = i

        for j = i + 1, n do
            if arr[j] < arr[min] then
                min = j
            end
        end

        arr[i], arr[min] = arr[min], arr[i]
    end
end

function main()
    io.write("Enter number of elements: ")
    local n = io.read("*n")

    local arr = {}

    for i = 1, n do
        io.write("Enter an element: ")
        arr[i] = io.read("*n")
    end

    selectionSort(arr)

    print("Sorted array: ")

    for i = 1, n do
        io.write(arr[i].." ")
    end
end

main()
